source/CCustomOpcodeSystem.cpp

```diff
@@ -738,20 +770,26 @@ namespace CLEO
 				case DT_LVAR_TEXTLABEL:
 				case DT_VAR_TEXTLABEL_ARRAY:
 				case DT_LVAR_TEXTLABEL_ARRAY:
-					return { (char*)GetScriptParamPointer(thread), 8 };
+					result.data = (char*)GetScriptParamPointer(thread);
+					result.size = 8;
+					result.needTerminator = false;
+					return result;
 
 				// long string variable
 				case DT_VAR_STRING:
 				case DT_LVAR_STRING:
 				case DT_VAR_STRING_ARRAY:
 				case DT_LVAR_STRING_ARRAY:
-					return { (char*)GetScriptParamPointer(thread), 16 };
+					result.data = (char*)GetScriptParamPointer(thread);
+					result.size = 16;
+					result.needTerminator = false;

```

Comment:
this should be result.needTerminator = true?

Reply:
@v can contain 16 characters, right?

Reply:
yes

Reply:
in this case terminator is not required (but it will be added for shorter texts)